test/v2/updateUKSavingsAccountName/UpdateUKSavingsAccountNameValidatorSpec.scala

```diff
@@ -0,0 +1,89 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v2.updateUKSavingsAccountName
+
+import models.domain.SavingsAccountId
+import models.errors.SavingsAccountIdFormatError
+import play.api.libs.json._
+import shared.models.domain.Nino
+import shared.models.errors._
+import shared.models.utils.JsonErrorValidators
+import shared.utils.UnitSpec
+import v2.updateUKSavingsAccountName.fixture.UpdateUKSavingsAccountNameFixtures.{requestBodyModel, validRequestJson}
+import v2.updateUKSavingsAccountName.model.request.UpdateUKSavingsAccountNameRequest
+
+class UpdateUKSavingsAccountNameValidatorSpec extends UnitSpec with JsonErrorValidators {
+
+  private implicit val correlationId: String = "1234"
+
+  private val validNino: String         = "AA123456A"
+  val validSavingsAccountId: String            = "SAVKB2UVwUTBQGJ"
+
+  private val parsedNino: Nino                 = Nino(validNino)
+  private val savingsAccountId = SavingsAccountId("SAVKB2UVwUTBQGJ")
+
+  private def validator(nino: String,
+                        savingsAccountId: String,
+                        body: JsValue): UpdateUKSavingsAccountNameValidator =
+    new UpdateUKSavingsAccountNameValidator(nino, savingsAccountId, body)
+
+  "running a validation" should {
+    "return no errors" when {
+      "a full valid request is supplied" in {
+        val result: Either[ErrorWrapper, UpdateUKSavingsAccountNameRequest] =
+          validator(validNino, validSavingsAccountId, validRequestJson).validateAndWrapResult()
+
+        result shouldBe Right(UpdateUKSavingsAccountNameRequest(parsedNino, savingsAccountId, requestBodyModel))
+      }
+    }
+
+    "return NinoFormatError error" when {
+      "an invalid nino is supplied" in {
+        val result: Either[ErrorWrapper, UpdateUKSavingsAccountNameRequest] =
+          validator("A12344A", validSavingsAccountId, validRequestJson).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, NinoFormatError))
+      }
+    }
+    "return SavingsAccountIdFormatError error" when {
+      "an invalid savings account id is supplied" in {
+        val result: Either[ErrorWrapper, UpdateUKSavingsAccountNameRequest] =
+          validator(validNino, "g434", validRequestJson).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, SavingsAccountIdFormatError))
+      }
+    }
+
+    "passed an empty body" in {
+      val result: Either[ErrorWrapper, UpdateUKSavingsAccountNameRequest] =
+        validator(validNino, validSavingsAccountId, JsObject.empty).validateAndWrapResult()
+
+      result shouldBe Left(ErrorWrapper(correlationId, RuleIncorrectOrEmptyBodyError))
+    }
+    }
```

Comment:
This curly bracket should be moved to the end of the test as the below test `return multiple errors` is not within `running a validation` so it is out of place. Please update, thanks.